### PR TITLE
DAOS-4778 test: Adding test for io consistency when accessed using different middleware

### PR DESCRIPTION
**Pull request by saurabhtandan**
Verify data consistency using different
middlewares. In this case, using DFS and POSIX.

As part of this PR two new files were added and
ior_test_base.py was modified.

Signed-off-by: Saurabh Tandan <saurabh.tandan@intel.com>

### CODE REVIEW

File: src/tests/ftest/io/io_consistency.py [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+            if api_flag[0] == "POSIX":
+                # if api is POSIX do not create new pool and container.
+                # Also do not stop dfuse when ior command is complete.
+                self.run_ior_with_pool(update=False, stop_dfuse=False)
+            else:
+                self.run_ior_with_pool()
+
+        # Try to re-create the file with same name and size.
+        self.ior_cmd.flags.update(apis_flags[0][1])
+        self.run_ior_with_pool(update=False)
```
Comment (shimizukko) [inline src/tests/ftest/io/io_consistency.py:75]: Don't you want to check the returned value? How do you know if the write fails? run_ior_with_pool doesn't raise any error.

File: src/tests/ftest/io/io_consistency.py [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+            super(IoConsistency, self).tearDown()
+
+    def test_ioconsistency(self):
+        """Jira ID: DAOS-4778.
+        Test Description:
+            Run IOR first using DFS and then using POSIX to verify data
+            consistency.
+        Use Cases:
+            Create a pool
+            Create POSIX type container.
```
Comment (shimizukko) [inline src/tests/ftest/io/io_consistency.py:47]: Maybe I'm just not knowledgeable enough.
Do you create the pool and the container through run_ior_with_pool? If so, what if api_flag[0] == "POSIX"? Can you run ior without a pool?
I don't know if this is the right PR to talk bout this, but I'm not sure if it's a good idea to combine pool create and run ior in the same method, especially with the update flag. Does the method work with update=False when the pool isn't created? It gets harder to understand and maintain if a method does two unrelated tasks.

File: src/tests/ftest/io/io_consistency.py [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+            Run ior -a DFS with FPP and keep the file.
+            Mount the container using dfuse
+            Run IOR -a POSIX -r -R to verify data consistency and delete the
+            file.
+            Try to re-write to the same file name after deletion, which should
```
Comment (shimizukko) [inline src/tests/ftest/io/io_consistency.py:52]: Is this the same step as your comment in Line 73? If so, why one is re-write and the other is re-create?

File: src/tests/ftest/io/io_consistency.yaml [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+    SSF:
+      api_flag:
+        - [DFS , "-v -w -k"]
+        - [POSIX, "-v -r -R"]
+  dfs_destroy: False
+  test_file: /testFile
+  signature: 5
+  repetitions: 1
+  transfersize_blocksize:
+    4K:
```
Comment (Puspalogan) [inline src/tests/ftest/io/io_consistency.yaml:50]: Change to 1M or remove this line as we're only doing one transfer_size.

File: src/tests/ftest/util/ior_test_base.py [hunk 5]
```diff
@@ -127,7 +127,7 @@ def _start_dfuse(self):
 
     def run_ior_with_pool(self, intercept=None, test_file_suffix="",
                           test_file="daos:testFile", update=True,
-                          create_cont=True):
```
Comment (Puspalogan) [inline src/tests/ftest/util/ior_test_base.py:130]: Rename update=True to create_pool=True, so you can do pool creation in conjunction with cont_create. As the signature of run_ior_with_pool grows, simple "update" wouldn't be clear for the user calling the method.

File: src/tests/ftest/io/io_consistency.py (hunk 3 above)
Comment (saurabhtandan) [reply src/tests/ftest/io/io_consistency.py]: It's kind of a same thing. But I can change it to 're-create' at both places.

File: src/tests/ftest/io/io_consistency.py (hunk 1 above)
Comment (saurabhtandan) [reply src/tests/ftest/io/io_consistency.py]: Nope. So `run_ior_with_pool()` calls `run_ior()`, which excepts `CommandFailure` and fail the test.

File: src/tests/ftest/io/io_consistency.yaml (hunk 4 above)
Comment (saurabhtandan) [reply src/tests/ftest/io/io_consistency.yaml:50]: Done

File: src/tests/ftest/util/ior_test_base.py (hunk 5 above)
Comment (saurabhtandan) [reply src/tests/ftest/util/ior_test_base.py:130]: Done

File: src/tests/ftest/io/io_consistency.py (hunk 1 above)
Comment (shimizukko) [reply src/tests/ftest/io/io_consistency.py]: I see. Thanks.

File: src/tests/ftest/io/io_consistency.py (hunk 2 above)
Comment (shimizukko) [reply src/tests/ftest/io/io_consistency.py:48]: Okay. Thanks for the explanation.

**Comment by saurabhtandan** [comment]
Test failures are not related to changes made in this PR

**Pull request merged**